core/domain/topic_fetchers.py

```diff
@@ -348,24 +382,28 @@ def get_published_topic_summaries():
         topic_id
         for topic_id, topic_rights in topic_id_to_topic_rights.items()
         if topic_rights.topic_is_published]
-    return get_multi_topic_summaries(published_topic_ids)
+    return [topic_summary for topic_summary in get_multi_topic_summaries(
+            published_topic_ids) if topic_summary is not None]
```

Comment:
To tell Mypy that the returned list doesn't contain `None` values.

Reply:
But this changes the behaviour, right? Previously it returned Nones too now it only returns non-nones

Reply:
Ahh. Nope, I guess as it was not returning `None` values. Mypy was just assuming the type as Optional because of the return type of the `get_multi_topic_summaries` function. And if we look into the `get_published_topic_summaries` function, we can see that it is only getting the summaries of the topics that are published which cancels out the probability of nonexistence of a topic ( in that case `get_multi_topic_summaries` will not return `None` ). does this answer your question?

Reply:
@SD-13 Sorry not sure what you mean.

Reply:
@vojtechjelinek  I mean, can we add an ignore here and revert the above change because from the code we can see that it is not returning `None` values. If have any suggestions then please let me know.

Reply:
I don't think we should use ignore here. We should try to either explain it or fix it a bit earlier in the calltree.